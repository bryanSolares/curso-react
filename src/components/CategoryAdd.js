import React, { useState } from "react";
import PropTypes from "prop-types";

export const CategoryAdd = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    }
  };

  return (
    <>
      <h2>Category Add</h2>
      <form onSubmit={handleSubmit}>
        <input type="text" value={inputValue} onChange={handleInputChange} />
      </form>
    </>
  );
};

CategoryAdd.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
