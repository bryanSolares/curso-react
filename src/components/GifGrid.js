import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  //   const [images, setImages] = useState([]);
  //   useEffect(() => {
  //     getGifs(category).then(setImages);
  //   }, [category]); // esto se implemento porque cada vez que el componente cambiaba se necesitaba recargar, se traslada a custome hook

  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn"> {category} </h3>

      {loading && <p className="animate__animated animate__flash">Cargando...</p>}

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
