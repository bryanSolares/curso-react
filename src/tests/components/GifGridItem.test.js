import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas con <GifGridItem />", () => {
  const title = "Un Titulo";
  const url = "https://localhost";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Debera mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener un parrafo con el title", () => {
    const parrafo = wrapper.find("p");
    expect(parrafo.text().trim()).toBe(title);
  });

  test("Debe tener una imagen, igual al url y alt de los props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe tener animate__fadeIn", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");
    expect(className.includes("animate__fadeIn")).toBeTruthy();
  });
});
