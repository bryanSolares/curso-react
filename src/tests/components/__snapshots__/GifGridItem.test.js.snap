// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifGridItem /> Debera mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un Titulo"
    src="https://localhost"
  />
  <p>
     
    Un Titulo
     
  </p>
</div>
`;
